src/syntree.rs

```diff
@@ -796,41 +1037,40 @@ mod expression {
                     };
                 }
 
+                // Parse return type
                 T::Arrow => {
                     ctx = ctx.skip(1);
                     break if let Ok((_ctx, ret)) = parse_type(ctx) {
-                        ctx = _ctx;
+                        ctx = _ctx; // assign to outer
                         ret
                     } else {
                         use RuntimeType::Void;
                         use TypeKind::Resolved;
                         Type {
+                            // If we couldn't parse the return type, we assume `-> Void`.
                             span: ctx.span(),
                             kind: Resolved(Void),
                         }
                     };
                 }
 
-                // TODO(ed): You're allowed to skip the arrow for void?
                 T::LeftBrace => {
                     use RuntimeType::Void;
                     use TypeKind::Resolved;
+                    // No return type so we assume `-> Void`.
                     break Type {
                         span: ctx.span(),
                         kind: Resolved(Void),
                     };
                 }
 
-                T::EOF => {
-                    raise_syntax_error!(ctx, "Didn't expect EOF in function");
-                }
-
                 t => {
                     raise_syntax_error!(ctx, "Didn't expect '{:?}' in function", t);
                 }
             }
         };
 
+        // Parse the function statement. Usually a block but it's not currently forced.
```

Comment:
```suggestion
        // Parse the function statement. Usually a block-statement but it's not currently forced.
```
There are multiple kinds of blocks - so it might help to specify which kind.